src/resources/js/v2/tickets-data.js

```diff
@@ -0,0 +1,268 @@
+	 *
+	 * @since TBD
+	 */
+	obj.clearLocal = function( eventId ) {
+		const postId = eventId || tribe.tickets.block.postId;
+
+		sessionStorage.removeItem( 'tribe_tickets_attendees-' + postId );
+		sessionStorage.removeItem( 'tribe_tickets_cart-' + postId );
+	};
+
+	/**
+	 * Gets attendee and cart form data from sessionStorage.
+	 *
+	 * @since TBD
+	 *
+	 * @param {number|string} eventId The ID of the event/post we're on.
+	 *
+	 * @returns {array} An array of the data.
+	 */
+	obj.getLocal = function( eventId ) {
+		const postId = eventId || tribe.tickets.block.postId;
+		const meta = window.JSON.parse( sessionStorage.getItem( 'tribe_tickets_attendees-' + postId ) );
+		const tickets = window.JSON.parse( sessionStorage.getItem( 'tribe_tickets_cart-' + postId ) );
+		const ret = {};
+		ret.meta = meta;
+		ret.tickets = tickets;
+
+		return ret;
+	};
+
+	/**
+	 * Get cart & meta data from sessionStorage, otherwise make an ajax call.
+	 * Always loads tickets from API on page load to be sure we keep up to date with the cart.
+	 *
+	 * This returns a deferred data object ( promise ) So when calling you need to use something like
+	 * jQuery's $.when()
+	 *
+	 * Example:
+	 *  $.when(
+	 *     obj.getData()
+	 *  ).then(
+	 *     function( data ) {
+	 *         // Do stuff with the data.
+	 *     }
+	 *  );
+	 *
+	 * @since TBD
+	 *
+	 * @param {boolean|string} pageLoad If we are experiencing a page load.
+	 *
+	 * @returns {object} Deferred data object.
+	 */
+	obj.getData = function( pageLoad ) {
+		let ret = {};
+		ret.meta = {};
+		ret.tickets = {};
+		const deferred = $.Deferred();
+		const meta = window.JSON.parse( sessionStorage.getItem( 'tribe_tickets_attendees-' + obj.postId ) );
+
+		if ( null !== meta ) {
+			ret.meta = meta;
+		}
+
+		// If we haven't reloaded the page, assume the cart hasn't changed since we did.
+		if ( ! pageLoad ) {
+			const tickets = window.JSON.parse( sessionStorage.getItem( 'tribe_tickets_cart-' + obj.postId ) );
+
+			if ( null !== tickets && tickets.length ) {
+				ret.tickets = tickets;
+			}
+
+			deferred.resolve( ret );
+		}
+
+		if ( ! ret.tickets || ! ret.meta ) {
+			$.ajax( {
+				type: 'GET',
+				data: {
+					provider: $tribeTicket.data( 'providerId' ),
+					post_id: tribe.tickets.block.postId,
+				},
+				dataType: 'json',
+				url: tribe.tickets.block.getRestEndpoint(),
+				success: function( data ) {
+					// Store for future use.
+					if ( null === meta ) {
+						sessionStorage.setItem(
+							'tribe_tickets_attendees-' + tribe.tickets.block.postId,
+							window.JSON.stringify( data.meta )
+						);
+					}
+
+					sessionStorage.setItem(
+						'tribe_tickets_cart-' + tribe.tickets.block.postId,
+						window.JSON.stringify( data.tickets )
+					);
+
+					ret = {
+						meta: data.meta,
+						tickets: data.tickets,
+					};
+
+					deferred.resolve( ret );
+				},
+				error: function() {
+					deferred.reject( false );
+				},
```

Comment:
Lets move into separate methods.

Reply:
This is not ready for review, pushed so Scott could know where we were at.